### PR TITLE
Udev vs tmpfiles take 3

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for https://github.com/systemd/systemd/commit/b42482af904ae0b94a6e4501ec595448f0ba1c06.

Fixes https://github.com/systemd/systemd/issues/28653 and https://github.com/systemd/systemd/issues/28765.
Replaces https://github.com/systemd/systemd/pull/28681 and https://github.com/systemd/systemd/pull/28732.

### CODE REVIEW

**Comment by yuwata** [comment]
(Huh, I was so stupid. Please kill me... I hope this fixes the original issue and the regression caused by the previous PR.)

**Review by bluca — APPROVED** [review]
makes sense to me

**Review by mrc0mmand — APPROVED** [review]
The original issue seems to be gone, thank you!

File: units/systemd-tmpfiles-setup-dev-early.service [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+[Unit]
+Description=Create Static Device Nodes in /dev gracefully
+Documentation=man:tmpfiles.d(5) man:systemd-tmpfiles(8)
+
+DefaultDependencies=no
+Before=sysinit.target local-fs-pre.target systemd-udevd.service
+Wants=local-fs-pre.target
+Conflicts=shutdown.target initrd-switch-root.target
+Before=shutdown.target initrd-switch-root.target
```
Comment (keszybz) [inline units/systemd-tmpfiles-setup-dev-early.service:18]: This should have `Before=systemd-tmpfiles-setup-dev.service`. This ordering exists via `systemd-sysusers.service`, but if that service is not active for whatever reason, we could end up with no ordering and a race between the two services.

**Review by keszybz — APPROVED** [review]
Looks very reasonable. Thanks for tracking this down.

**Comment by ElvishJerricco** [comment]
Glad I found this! Thanks. Will this make it into a 254 release?

**Comment by yuwata** [comment]
Yeah, this should be in the next point release, unless any issue will be found in this patch.

**Comment by keszybz** [comment]
Queued up for v254.2. Fingers crossed.

**Pull request merged**